src/client/src/Stream.c

```diff
@@ -1598,14 +1600,22 @@ STATUS waitForAvailability(PKinesisVideoStream pKinesisVideoStream, UINT32 alloc
         // Early return if available
         CHK(!IS_VALID_ALLOCATION_HANDLE(*pAllocationHandle), STATUS_SUCCESS);
 
-        // if no space available, wait only if in OFFLINE mode
-        CHK(IS_OFFLINE_STREAMING_MODE(pKinesisVideoStream->streamInfo.streamCaps.streamingType), STATUS_SUCCESS);
+        // if no space available, wait only if in OFFLINE mode or if we are going to evict the tail frames
+        CHK(pKinesisVideoStream->streamInfo.streamCaps.storePressurePolicy == CONTENT_STORE_PRESSURE_POLICY_DROP_TAIL_ITEM ||
+            IS_OFFLINE_STREAMING_MODE(pKinesisVideoStream->streamInfo.streamCaps.streamingType), STATUS_SUCCESS);
 
-        // Long path which will await for the availability notification or cancellation
-        CHK_STATUS(pKinesisVideoClient->clientCallbacks.waitConditionVariableFn(pKinesisVideoClient->clientCallbacks.customData,
-                                                                                pKinesisVideoStream->bufferAvailabilityCondition,
-                                                                                pKinesisVideoStream->base.lock,
-                                                                                pKinesisVideoClient->deviceInfo.clientInfo.offlineBufferAvailabilityTimeout));
+        if (IS_OFFLINE_STREAMING_MODE(pKinesisVideoStream->streamInfo.streamCaps.streamingType)) {
+            // Long path which will await for the availability notification or cancellation
+            CHK_STATUS(pKinesisVideoClient->clientCallbacks.waitConditionVariableFn(
+                    pKinesisVideoClient->clientCallbacks.customData,
+                    pKinesisVideoStream->bufferAvailabilityCondition,
+                    pKinesisVideoStream->base.lock,
+                    pKinesisVideoClient->deviceInfo.clientInfo.offlineBufferAvailabilityTimeout));
+        } else {
+            // Need to evict the tail frames by trimming the tail
+            CHK_STATUS(contentViewGetTail(pKinesisVideoStream->pView, &pViewItem));
```

Comment:
this behavior should also be affected by viewOverflowPolicy (drop one item vs drop fragment.)

Reply:
Yes indeed. For some reason I entirely forgot the reason I rebased this on top of your change :) Will send an update